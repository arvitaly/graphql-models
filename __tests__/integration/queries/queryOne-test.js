"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const utils_1 = require("./../utils");
it("query one: animal", () => __awaiter(this, void 0, void 0, function* () {
    const tester = new utils_1.default();
    const animal = tester.generateModel("animal");
    tester.adapter.findOne.mockReturnValueOnce(animal);
    const globalId = tester.toGlobalId("Animal", animal.id);
    const result = yield tester.query(`query Q1{
            viewer{
                animal(id:"${globalId}"){
                    id
                    name
                    age
                    birthday
                    some
                    Weight
                    isCat
                }
            }
        }`);
    expect(result).toMatchSnapshot();
}));
it("query one: post", () => __awaiter(this, void 0, void 0, function* () {
    const tester = new utils_1.default();
    const animal = tester.generateModel("animal");
    const animal2 = tester.generateModel("animal");
    const user = tester.generateModel("user", { pets: [animal2] });
    const post = tester.generateModel("post", {
        animals: [animal],
        owner: user,
    });
    tester.adapter.findOne.mockImplementation((modelId) => {
        switch (modelId) {
            case "post":
                return post;
            case "animal":
                return animal;
        }
    });
    const result = yield tester.query(`query Q1{
            viewer{
                post(id: "${tester.toGlobalId("Post", post.id)}"){
                    owner{
                        name
                        pets{
                            edges{
                                node{
                                    ...F1
                                }
                            }
                        }
                    }
                    animals{
                        edges{
                            node{
                                ...F1
                            }
                        }
                    }
                }
            }
        }
        fragment F1 on Animal{
            id
            name
            age
            birthday
            some
            Weight
            isCat
        }`);
    expect(result).toMatchSnapshot();
}));
