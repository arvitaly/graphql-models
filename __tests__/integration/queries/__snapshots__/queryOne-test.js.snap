// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query one: animal 1`] = `
Object {
  "viewer": Object {
    "animal": Object {
      "Weight": 1001.1,
      "age": 101,
      "birthday": "Fri, 01 Jan 2010 00:00:00 GMT",
      "id": "QW5pbWFsOjE=",
      "isCat": true,
      "name": "Str1",
      "some": Object {
        "test": 1,
      },
    },
  },
}
`;

exports[`query one: post 1`] = `
Object {
  "viewer": Object {
    "post": Object {
      "animals": Object {
        "edges": Array [
          Object {
            "node": Object {
              "Weight": 1001.1,
              "age": 101,
              "birthday": "Fri, 01 Jan 2010 00:00:00 GMT",
              "id": "QW5pbWFsOjE=",
              "isCat": true,
              "name": "Str1",
              "some": Object {
                "test": 1,
              },
            },
          },
        ],
      },
      "owner": Object {
        "name": "Str1",
        "pets": Object {
          "edges": Array [
            Object {
              "node": Object {
                "Weight": 1002.1,
                "age": 102,
                "birthday": "Fri, 01 Jan 2010 00:00:00 GMT",
                "id": "QW5pbWFsOjI=",
                "isCat": true,
                "name": "Str2",
                "some": Object {
                  "test": 2,
                },
              },
            },
          ],
        },
      },
    },
  },
}
`;
